blocks/library/paragraph/index.js

```diff
@@ -25,8 +25,186 @@ import ToggleControl from '../../inspector-controls/toggle-control';
 import BlockDescription from '../../block-description';
+import ContrastChecker from '../../contrast-checker';
 
 const { children } = source;
+const { getComputedStyle } = window;
+
+class ParagraphBlock extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.editableRef = null;
+
+		this.state = {
+			fallbackTextColor: undefined,
+		};
+
+		this.bindRef = this.bindRef.bind( this );
+		this.toggleDropCap = this.toggleDropCap.bind( this );
+	}
+
+	toggleDropCap() {
+		const { attributes, setAttributes } = this.props;
+		setAttributes( { dropCap: ! attributes.dropCap } );
+	}
+
+	bindRef( node ) {
+		if ( ! node ) {
+			return;
+		}
+		this.editableRef = node.querySelector( '[contenteditable="true"]' );
+	}
+
+	componentDidMount() {
+		this.grabTextColor();
+	}
+
+	componentDidUpdate() {
+		this.grabTextColor();
+	}
+
+	grabTextColor() {
+		const { textColor } = this.props.attributes;
+		const { fallbackTextColor } = this.state;
+
+		if ( ! textColor && ! fallbackTextColor && this.editableRef ) {
+			this.setState( { fallbackTextColor: getComputedStyle( this.editableRef ).color } );
+		}
+	}
+
+	render() {
+		const {
+			attributes,
+			setAttributes,
+			insertBlocksAfter,
+			focus,
+			setFocus,
+			mergeBlocks,
+			onReplace,
+		} = this.props;
+
+		const {
+			align,
+			content,
+			dropCap,
+			placeholder,
+			fontSize,
+			backgroundColor,
+			textColor,
+			width,
+		} = attributes;
+
+		const {
+			fallbackTextColor,
+		} = this.state;
+
+		const className = dropCap ? 'has-drop-cap' : null;
+
+		return [
+			focus && (
+				<BlockControls key="controls">
+					<AlignmentToolbar
+						value={ align }
+						onChange={ ( nextAlign ) => {
+							setAttributes( { align: nextAlign } );
+						} }
+					/>
+				</BlockControls>
+			),
+			focus && (
+				<InspectorControls key="inspector">
+					<BlockDescription>
+						<p>{ __( 'Text. Great things start here.' ) }</p>
+					</BlockDescription>
+					<PanelBody title={ __( 'Text Settings' ) }>
+						<ToggleControl
+							label={ __( 'Drop Cap' ) }
+							checked={ !! dropCap }
+							onChange={ this.toggleDropCap }
+						/>
+						<RangeControl
+							label={ __( 'Font Size' ) }
+							value={ fontSize || '' }
+							onChange={ ( value ) => setAttributes( { fontSize: value } ) }
+							min={ 10 }
+							max={ 200 }
+							beforeIcon="editor-textcolor"
+							allowReset
+						/>
+					</PanelBody>
+					<PanelBody title={ __( 'Background Color' ) }>
+						<ColorPalette
+							value={ backgroundColor }
+							onChange={ ( colorValue ) => setAttributes( { backgroundColor: colorValue } ) }
+						/>
+					</PanelBody>
+					<PanelBody title={ __( 'Text Color' ) }>
+						<ColorPalette
+							value={ textColor }
+							onChange={ ( colorValue ) => setAttributes( { textColor: colorValue } ) }
+						/>
+					</PanelBody>
+					<ContrastChecker
+						textColor={ textColor || fallbackTextColor }
+						backgroundColor={ backgroundColor || '#fff' }
```

Comment:
Is there a plan to add fallback logic to `backgroundColor`, too? Or maybe it is always white anyway :)

Reply:
That's a nice question, I used a logic for background color but, the getComputeStyle always returned black, while the background is white If background color was explicitly set the getComputedSyle returned correctly. So I assumed the background when not explicitly set is white.